packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,384 @@
+import { vec3 } from 'gl-matrix';
+import {
+  metaData,
+  getRenderingEngines,
+  CONSTANTS,
+  utilities as csUtils,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+
+import { addAnnotation } from '../stateManagement/annotation/annotationState';
+
+import { drawLine as drawLineSvg } from '../drawingSvg';
+import { filterViewportsWithToolEnabled } from '../utilities/viewportFilters';
+import triggerAnnotationRenderForViewportIds from '../utilities/triggerAnnotationRenderForViewportIds';
+
+import { PublicToolProps, ToolProps, SVGDrawingHelper } from '../types';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import AnnotationDisplayTool from './base/AnnotationDisplayTool';
+import { Annotation } from '../types';
+import { render } from 'react-dom';
+
+const { EPSILON } = CONSTANTS;
+
+/**
+ * Get the largest axis of a normal vector. We assume that indicates the image
+ * orientation ( AXIAL = 2, CORONAL = 1, SAGITTAL = 0)
+ * @param normalVector
+ * @returns
+ */
+function getNormalLargestAxis(normalVector) {
+  let largestAxis = 0;
+  for (let i = 1; i < normalVector.length; i++) {
+    if (Math.abs(normalVector[i]) > Math.abs(normalVector[largestAxis])) {
+      largestAxis = i;
+    }
+  }
+  return largestAxis;
+}
+
+/**
+ * Get the largest normal axis of the a viewport first imageId
+ * @param viewport
+ * @returns
+ */
+function getViewportImageIdNormalAxis(viewport) {
+  const imageIds = viewport.getImageIds();
+  const { rowCosines, columnCosines } = metaData.get(
+    'imagePlaneModule',
+    imageIds[0]
+  );
+  const scanAxisNormal = vec3.cross(vec3.create(), rowCosines, columnCosines);
+  return getNormalLargestAxis(scanAxisNormal);
+}
+
+export interface OverlayGridAnnotation extends Annotation {
+  data: {
+    sourceViewport;
+  };
+}
+
+/**
+ * @public
+ */
+class OverlayGridTool extends AnnotationDisplayTool {
+  static toolName;
+
+  public touchDragCallback: any;
+  public mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    renderingEngine: any;
+    annotations: Map<string, Annotation>;
+  } | null = {} as any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        sourceViewportIds: [],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  _init = (): void => {
+    const renderingEngines = getRenderingEngines();
+    const renderingEngine = renderingEngines[0];
+
+    // Todo: handle this case where it is too soon to get the rendering engine
+    if (!renderingEngine) {
+      return;
+    }
+
+    let viewports = renderingEngine.getViewports();
+    viewports = filterViewportsWithToolEnabled(viewports, this.getToolName());
+
```

Comment:
it should act on ONLY on the viewports from this toolGroup. You have access to this.toolGroup